lib/galaxy/webapps/tool_shed/api/categories.py

```diff
@@ -103,4 +107,6 @@ def show( self, trans, id, **kwd ):
         category_dict[ 'url' ] = web.url_for( controller='categories',
                                               action='show',
                                               id=trans.security.encode_id( category.id ) )
+        if util.asbool( kwd.get( 'show_repositories', False ) ):
+            category_dict[ 'repositories' ] = suc.get_repositories_by_category( trans.app, category.id )
```

Comment:
similar to L86 above - this is better but separate endpoint would be the best I believe